spec.proto

```diff
@@ -0,0 +1,55 @@
+/**
+ * CloudEvent Protobuf Format
+ *
+ * - Required context attributes are explicity represented.
+ * - Optional and Extension context attributes are carried in a map structure.
+ * - Data may be represented as binary, text, or protobuf messages.
+ */
+
+syntax = "proto3";
+
+package io.cloudevents.v1;
+
+import "google/protobuf/any.proto";
+import "google/protobuf/timestamp.proto";
+
+option java_package = "io.cloudevents.v1.proto";
+
+message CloudEvent {
+
+  // -- CloudEvent Context Attributes
+
+  // Required Attributes
+  string id = 1;
+  string source = 2; // URI-reference
+  string spec_version = 3;
+  string type = 4;
+
+  // Optional & Extension Attributes
+  map<string, CloudEventAttribute> attribute = 5;
```

Comment:
```suggestion
  map<string, CloudEventAttribute> attributes = 5;
```

(Old Google naming conventions used singular names even for repeated fields and maps... we've moved to plural names now.)

Reply:
We moved from plural to a singular name as the generated code read more naturally ... I'm happy either way, I think we just need to pick one.

If plurals are now part of a style-guide then we should switch back.

@evankanderson

Reply:
I'm fine either way.